### PR TITLE
refactor: adding semantic release for v21

### LINKED ISSUE #146: refactor: Github release version does not match package.json versions
All releases in the ts-sdk should share the same number, this is a developer necessity, so they don't have to guess if they have the right versions

Also we should start releasing to github with releases and/or tags

### LINKED ISSUE #147: feat: Semantic release and automatic changelog generation ts-sdk

### PR DESCRIPTION
**Pull request by CalicoNino**
- Closes #146 
- Closes #147 

Tested in dummy [repo](https://github.com/elshenak/testing/releases)
- https://www.npmjs.com/package/@elshenak/indexer-nibi
- https://www.npmjs.com/package/@elshenak/nibijs
- https://www.npmjs.com/package/@elshenak/protojs

### CODE REVIEW

**Comment by CalicoNino** [comment]
Will need to update NPM_TOKEN and GITHUB_TOKEN

**Comment by cgilbe27** [comment]
Tests failing

**Review by cgilbe27 — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -27,7 +27,8 @@
     "format:md": "yarn lint:md --fix",
     "proto-gen": "yarn exec ./scripts/protocgen.sh",
     "prettier": "prettier --write . --ignore-path .gitignore",
-    "prepare": "husky install"
+    "prepare": "husky install",
+    "semantic-release": "lerna exec --concurrency 1 -- semantic-release -e semantic-release-monorepo --tag-format='${LERNA_PACKAGE_NAME}-v\\${version}'"
```
Comment (cgilbe27) [inline package.json:31]: Why are we limiting concurrency?

Comment (CalicoNino) [reply package.json:31]: No specific reason, just followed their documentation https://github.com/pmowrer/semantic-release-monorepo#tagformat

Comment (CalicoNino) [reply package.json:31]: don't think it will cause an issue as I tested with this concurrency limit

**Pull request merged**